src/overlays/actors/ovl_En_Box/z_en_box.c

```diff
@@ -73,14 +73,25 @@ static InitChainEntry sInitChain[] = {
 };
 
 typedef struct {
-    /* 0x0 */ f32 data[5];
+    /* 0x0 */ f32 data[PLAYER_FORM_MAX];
 } EnBox_PlaybackSpeed; // 0x14
 
```

Comment:
So I'm not 100% sure, but you can remove the struct and struct copy if you declare the `sPlaybackSpeed` array in function like so. (Note the animations array must be in function as well to preserve ordering, but can be declared at the top or in the same branch as the speed table)

```c
void EnBox_WaitOpen(EnBox* this, PlayState* play) {
   static AnimationHeader* sBigChestAnimations[PLAYER_FORM_MAX] = {
       &gBoxBigChestOpenAdultAnim, // PLAYER_FORM_FIERCE_DEITY
       &gBoxBigChestOpenGoronAnim, // PLAYER_FORM_GORON
       &gBoxBigChestOpenAdultAnim, // PLAYER_FORM_ZORA
       &gBoxBigChestOpenDekuAnim,  // PLAYER_FORM_DEKU
       &gBoxBigChestOpenChildAnim, // PLAYER_FORM_HUMAN
    };
    s32 pad;
    AnimationHeader* animHeader;
    f32 frameCount;
    f32 playbackSpeed;

    this->alpha = 255;
    this->movementFlags |= ENBOX_MOVE_IMMOBILE;
    if ((this->unk_1EC != 0) && ((this->csId2 < 0) || (CutsceneManager_GetCurrentCsId() == this->csId2) ||
                                 (CutsceneManager_GetCurrentCsId() == CS_ID_NONE))) {
        if (this->unk_1EC < 0) {
            animHeader = &gBoxChestOpenAnim;
            playbackSpeed = 1.5f;
        } else {
            f32 playbackSpeedTable[] = {
                1.5f, // PLAYER_FORM_FIERCE_DEITY
                1.0f, // PLAYER_FORM_GORON
                1.5f, // PLAYER_FORM_ZORA
                1.0f, // PLAYER_FORM_DEKU
                1.5f, // PLAYER_FORM_HUMAN
            };

            animHeader = sBigChestAnimations[GET_PLAYER_FORM];
            playbackSpeed = playbackSpeedsTable[GET_PLAYER_FORM];
        }

        frameCount = Animation_GetLastFrame(animHeader);
        Animation_Change(&this->skelAnime, animHeader, playbackSpeed, 0.0f, frameCount, ANIMMODE_ONCE, 0.0f);
        EnBox_SetupAction(this, EnBox_Open);
        if (this->unk_1EC > 0) {
            Actor_SpawnAsChild(&play->actorCtx, &this->dyna.actor, play, ACTOR_DEMO_TRE_LGT,
                               this->dyna.actor.world.pos.x, this->dyna.actor.world.pos.y, this->dyna.actor.world.pos.z,
                               this->dyna.actor.shape.rot.x, this->dyna.actor.shape.rot.y, this->dyna.actor.shape.rot.z,
                               -1);
            Audio_PlayFanfare(NA_BGM_OPEN_CHEST | 0x900);
        }

        if (this->getItemId == GI_STRAY_FAIRY) {
            this->movementFlags |= ENBOX_MOVE_0x20;
        } else {
            if ((this->getItemId == GI_HEART_PIECE) || (this->getItemId == GI_BOTTLE)) {
                Flags_SetCollectible(play, this->collectableFlag);
            }
            Flags_SetTreasure(play, ENBOX_GET_CHEST_FLAG(&this->dyna.actor));
        }
    } else {
        Player* player = GET_PLAYER(play);
        Vec3f offset;

        Actor_OffsetOfPointInActorCoords(&this->dyna.actor, &offset, &player->actor.world.pos);
        if ((offset.z > -50.0f) && (offset.z < 0.0f) && (fabsf(offset.y) < 10.0f) && (fabsf(offset.x) < 20.0f) &&
            Player_IsFacingActor(&this->dyna.actor, 0x3000, play)) {
            if (((this->getItemId == GI_HEART_PIECE) || (this->getItemId == GI_BOTTLE)) &&
                Flags_GetCollectible(play, this->collectableFlag)) {
                this->getItemId = GI_RECOVERY_HEART;
            }
            if ((this->getItemId == GI_MASK_CAPTAIN) && (INV_CONTENT(ITEM_MASK_CAPTAIN) == ITEM_MASK_CAPTAIN)) {
                this->getItemId = GI_RECOVERY_HEART;
            }
            if ((this->getItemId == GI_MASK_GIANT) && (INV_CONTENT(ITEM_MASK_GIANT) == ITEM_MASK_GIANT)) {
                this->getItemId = GI_RECOVERY_HEART;
            }
            Actor_OfferGetItemNearby(&this->dyna.actor, play, -this->getItemId);
        }
        if (Flags_GetTreasure(play, ENBOX_GET_CHEST_FLAG(&this->dyna.actor))) {
            EnBox_SetupAction(this, EnBox_Open);
        }
    }
}
```

Reply:
However if we do keep the struct it should be `EnBoxPlaybackSpeeds` (no _ and add an extra `s`)

Reply:
I do like it better without the struct, nice find. The data is right there and you don't need to manage any tables or weird structs